instaloader/__main__.py

```diff
@@ -66,6 +67,45 @@ def filterfunc(item) -> bool:
 
     return filterfunc
 
+def get_cookies_from_instagram(domain, browser, cookiefile="", cookieName=''):
+    supported_browsers = {
+        "chrome": browser_cookie3.chrome,
+        "firefox": browser_cookie3.firefox,
+        "edge": browser_cookie3.edge,
+        "brave": browser_cookie3.brave,
+        "opera": browser_cookie3.opera,
+        "safari": browser_cookie3.safari
+    }
+
+    if browser not in supported_browsers:
+        print("Loading cookies from the specified browser failed. Supported browsers are Chrome, Firefox, Edge, Brave, Opera and Safari")
+        return {}
+
+    Cookies = {}
+    browserCookies = list(supported_browsers[browser](cookie_file=cookiefile))
+
+    for cookie in browserCookies:
+        if domain in cookie.domain:
+            Cookies[cookie.name] = cookie.value
+
+    if Cookies:
+        print(f"Cookies loaded successfully from {browser}")
+    else:
+        print(f"No cookies found for Instagram in {browser}, Are you logged in succesfully in {browser}?")
+
+    if cookieName:
+        return Cookies.get(cookieName, {})
+    else:
+        return Cookies
+
+def import_session(browser, instaloader, cookiefile):
+    cookie = get_cookies_from_instagram('instagram', browser, cookiefile)
+    if cookie is not None:
+        instaloader.context._session.cookies.update(cookie)
+        username = instaloader.test_login()
+        if not username:
+            raise SystemExit(f"Not logged in. Are you logged in successfully in {browser}?")
+        instaloader.context.username = username
```

Comment:
This function mostly accesses properties of InstaloaderContext, one of them even a protected property. Maybe we could just move it to being a method of InstaloaderContext?

Reply:
The `_session` object is not supposed to be accessed from outside of the InstaloaderContext.

Reply:
Shall I create a method in InstaloaderContext that returns the `_session`? would that be fine?

for example somthing like:

```python
def get_session(self):
        return self._session
```

then maybe in the `import_session` function I can do something like the following to update cookies:

```python
session = instaloader.context.get_session()
session.cookies.update(cookie)
```

In this example the `get_session` method is just the public method (or wrapper one might say) to get the session from InstaloaderContext. This avoid using session directly in InstaloaderContext.

Reply:
The session object should be kept as private as possible to the InstaloaderContext, if we would allow to manipulate it from outside we would invite many strange bug reports. We would really favor to move the part that updates the login cookie to InstaloaderContext.